pandas/io/pickle.py

```diff
@@ -63,8 +76,12 @@ def to_pickle(obj, path, compression="infer", protocol=pickle.HIGHEST_PROTOCOL):
     >>> import os
     >>> os.remove("./dummy.pkl")
     """
-    path = stringify_path(path)
-    f, fh = get_handle(path, "wb", compression=compression, is_text=False)
+    fp_or_buf, _, compression, should_close = get_filepath_or_buffer(
+        filepath_or_buffer, compression=compression, mode="wb"
+    )
+    if not isinstance(fp_or_buf, str) and compression == "infer":
```

Comment:
is this necessary? get_handle i believe does this already

Reply:
I think it is, [_get_handle_ calls infer_compression only when _path_or_buffer_ is a string](https://github.com/pandas-dev/pandas/blob/master/pandas/io/common.py#L388-L389).
So if we hand over a buffer, _compression_ stays as "infer", and later on, [it'll raise ValueError if the compression is not "gzip", "bz2", "zip", nor "xz".](https://github.com/pandas-dev/pandas/blob/master/pandas/io/common.py#L391-L433)